src/vm/types/mod.rs

```diff
@@ -253,6 +253,96 @@ impl SequenceData {
         }
     }
 
+    pub fn element_at(self, index: usize) -> Option<Value> {
+        if self.len() <= index {
+            return None;
+        }
+        let result = match self {
+            SequenceData::Buffer(data) => Value::buff_from_byte(data.data[index]),
+            SequenceData::List(mut data) => data.data.remove(index),
+            SequenceData::String(CharType::ASCII(data)) => {
+                Value::string_ascii_from_bytes(vec![data.data[index]])
+                    .expect("BUG: failed to initialize single-byte ASCII buffer")
+            }
+            SequenceData::String(CharType::UTF8(mut data)) => {
+                Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data {
+                    data: vec![data.data.remove(index)],
+                })))
+            }
+        };
+
+        Some(result)
+    }
+
+    pub fn contains(&self, to_find: Value) -> Result<Option<usize>> {
+        match self {
+            SequenceData::Buffer(ref data) => {
+                if let Value::Sequence(SequenceData::Buffer(to_find_vec)) = to_find {
+                    if to_find_vec.data.len() != 1 {
+                        Ok(None)
```

Comment:
Is this reachable?  Doesn't the type-checker already ensure that `to_find` has length 1?

Reply:
Same question for the other cases below.

Reply:
Yes, it is -- the type system only operates on the maximum possible length of the list, so a type signature of `(list 1 uint)` would admit an empty list (this is true of our buffer and string types as well).